From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Wed, 22 Aug 2012 20:30:09 +0200
Subject: [PATCH] wait for mmaped buffers, when needed

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2bufferpool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2bufferpool.c b/sys/v4l2/gstv4l2bufferpool.c
index 648217a..987dec2 100644
--- a/sys/v4l2/gstv4l2bufferpool.c
+++ b/sys/v4l2/gstv4l2bufferpool.c
@@ -1350,7 +1350,7 @@ gst_v4l2_buffer_pool_process (GstV4l2BufferPool * pool, GstBuffer * buf)
             if (!start_streaming (pool))
               goto start_failed;
 
-          if (pool->num_queued == pool->num_allocated) {
+          if (pool->num_queued > pool->obj->min_queued) {
             GstBuffer *out;
             /* all buffers are queued, try to dequeue one and release it back
              * into the pool so that _acquire can get to it again. */
