From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 28 May 2013 19:14:51 +0200
Subject: [PATCH] v4l2: better names for the bufferpools

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2bufferpool.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sys/v4l2/gstv4l2bufferpool.c b/sys/v4l2/gstv4l2bufferpool.c
index 71ffe22..4cc468f 100644
--- a/sys/v4l2/gstv4l2bufferpool.c
+++ b/sys/v4l2/gstv4l2bufferpool.c
@@ -1151,6 +1151,7 @@ gst_v4l2_buffer_pool_new (GstV4l2Object * obj, GstCaps * caps)
   GstV4l2BufferPool *pool;
   GstStructure *s;
   gint fd;
+  gchar *name;
 
   fd = v4l2_dup (obj->video_fd);
   if (fd < 0)
@@ -1161,6 +1162,12 @@ gst_v4l2_buffer_pool_new (GstV4l2Object * obj, GstCaps * caps)
   pool->obj = obj;
   pool->can_alloc = TRUE;
 
+  name = g_strdup_printf ("%s-%s-bufferpool",
+      gst_object_get_name (GST_OBJECT (obj->element)),
+      (obj->type == V4L2_BUF_TYPE_VIDEO_CAPTURE) ? "src" : "sink");
+  gst_object_set_name (GST_OBJECT (pool), name);
+  g_free (name);
+
   s = gst_buffer_pool_get_config (GST_BUFFER_POOL_CAST (pool));
   gst_buffer_pool_config_set_params (s, caps, obj->sizeimage, 2, 0);
   gst_buffer_pool_set_config (GST_BUFFER_POOL_CAST (pool), s);
