From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 12 Sep 2013 16:56:56 +0200
Subject: [PATCH] v4l2: set GST_BUFFER_FLAG_DELTA_UNIT when appropriate

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2bufferpool.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sys/v4l2/gstv4l2bufferpool.c b/sys/v4l2/gstv4l2bufferpool.c
index 4cc468f..45e952b 100644
--- a/sys/v4l2/gstv4l2bufferpool.c
+++ b/sys/v4l2/gstv4l2bufferpool.c
@@ -821,6 +821,12 @@ gst_v4l2_buffer_pool_dqbuf (GstV4l2BufferPool * pool, GstBuffer ** buffer)
   if (vbuffer.field == V4L2_FIELD_INTERLACED_BT) {
     GST_BUFFER_FLAG_UNSET (outbuf, GST_VIDEO_BUFFER_FLAG_TFF);
   }
+  if (GST_VIDEO_INFO_FORMAT (&obj->info) == GST_VIDEO_FORMAT_ENCODED) {
+    if (vbuffer.flags & V4L2_BUF_FLAG_KEYFRAME)
+      GST_BUFFER_FLAG_UNSET (outbuf, GST_BUFFER_FLAG_DELTA_UNIT);
+    else
+      GST_BUFFER_FLAG_SET (outbuf, GST_BUFFER_FLAG_DELTA_UNIT);
+  }
 
   /* The size can change at every frame, esp. with jpeg. The GstMemory
    * inside the GstBuffer could have been changed by some other
