From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 19 Apr 2012 11:23:29 +0200
Subject: [PATCH] v4l2: add gst_v4l2_object_start_shared

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2object.c | 15 +++++++++++++++
 sys/v4l2/gstv4l2object.h |  2 ++
 2 files changed, 17 insertions(+)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 7cd95d6..c18ed65 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -912,6 +912,21 @@ gst_v4l2_object_open (GstV4l2Object * v4l2object)
 }
 
 gboolean
+gst_v4l2_object_open_shared (GstV4l2Object * v4l2object, GstV4l2Object * other)
+{
+  if (gst_v4l2_clone (v4l2object, other))
+    gst_v4l2_set_defaults (v4l2object);
+  else
+    return FALSE;
+
+#ifdef HAVE_XVIDEO
+  gst_v4l2_video_overlay_start (v4l2object);
+#endif
+
+  return TRUE;
+}
+
+gboolean
 gst_v4l2_object_close (GstV4l2Object * v4l2object)
 {
 #ifdef HAVE_XVIDEO
diff --git a/sys/v4l2/gstv4l2object.h b/sys/v4l2/gstv4l2object.h
index 100cb3e..3c993fb 100644
--- a/sys/v4l2/gstv4l2object.h
+++ b/sys/v4l2/gstv4l2object.h
@@ -210,6 +210,8 @@ gboolean  gst_v4l2_object_get_property_helper       (GstV4l2Object *v4l2object,
 						     GParamSpec * pspec);
 /* open/close */
 gboolean  gst_v4l2_object_open               (GstV4l2Object *v4l2object);
+gboolean  gst_v4l2_object_open_shared        (GstV4l2Object *v4l2object,
+                                              GstV4l2Object * other);
 gboolean  gst_v4l2_object_close              (GstV4l2Object *v4l2object);
 
 /* probing */
