From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Thu, 30 May 2013 17:06:00 +0200
Subject: [PATCH] v4l2: filter: send V4L2_DEC_CMD_STOP on EOS

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2filter.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/sys/v4l2/gstv4l2filter.c b/sys/v4l2/gstv4l2filter.c
index 05cca49..523f2a0 100644
--- a/sys/v4l2/gstv4l2filter.c
+++ b/sys/v4l2/gstv4l2filter.c
@@ -463,6 +463,7 @@ static gboolean
 gst_v4l2filter_sink_event (GstPad * pad, GstObject * parent, GstEvent * event)
 {
   GstV4l2Filter *v4l2filter = GST_V4L2FILTER (parent);
+  struct v4l2_decoder_cmd dcmd;
   gboolean ret = TRUE;
 
   switch (GST_EVENT_TYPE (event)) {
@@ -475,6 +476,19 @@ gst_v4l2filter_sink_event (GstPad * pad, GstObject * parent, GstEvent * event)
       gst_event_unref (event);
       break;
     }
+    case GST_EVENT_EOS:
+      GST_LOG_OBJECT (v4l2filter, "got EOS event, stopping decoder");
+      /*
+       * Send V4L2_DEC_CMD_STOP decoder command on the v4l2 output side and
+       * continue to dequeue remaining buffers at the v4l2 capture side.
+       */
+      memset (&dcmd, 0, sizeof (dcmd));
+      dcmd.cmd = V4L2_DEC_CMD_STOP;
+      dcmd.flags = 0;
+      dcmd.stop.pts = 0;
+      v4l2_ioctl (v4l2filter->sink_v4l2object->video_fd,
+          VIDIOC_DECODER_CMD, &dcmd);
+      /* fallthrough */
     default:
       g_mutex_lock (&v4l2filter->queue_lock);
       if (GST_EVENT_IS_SERIALIZED (event)
