diff -Naur aa/sys/v4l2/gstv4l2videodec.c bb/sys/v4l2/gstv4l2videodec.c
--- aa/sys/v4l2/gstv4l2videodec.c	2016-06-08 12:53:02.000000000 +0200
+++ bb/sys/v4l2/gstv4l2videodec.c	2018-03-22 10:47:49.514923391 +0100
@@ -613,20 +613,20 @@
     GST_DEBUG_OBJECT (self, "Possible decoded caps: %" GST_PTR_FORMAT, caps);
     if (gst_caps_is_empty (caps)) {
       gst_caps_unref (caps);
-      goto not_negotiated;
     }
+    else {
+	/* Fixate pixel format */
+	caps = gst_caps_fixate (caps);
 
-    /* Fixate pixel format */
-    caps = gst_caps_fixate (caps);
+	GST_DEBUG_OBJECT (self, "Chosen decoded caps: %" GST_PTR_FORMAT, caps);
 
-    GST_DEBUG_OBJECT (self, "Chosen decoded caps: %" GST_PTR_FORMAT, caps);
-
-    /* Try to set negotiated format, on success replace acquired format */
-    if (gst_v4l2_object_set_format (self->v4l2capture, caps, &error))
-      gst_video_info_from_caps (&info, caps);
-    else
-      gst_v4l2_clear_error (&error);
-    gst_caps_unref (caps);
+	/* Try to set negotiated format, on success replace acquired format */
+	if (gst_v4l2_object_set_format (self->v4l2capture, caps, &error))
+		gst_video_info_from_caps (&info, caps);
+	else
+		gst_v4l2_clear_error (&error);
+	gst_caps_unref (caps);
+    }
 
     output_state = gst_video_decoder_set_output_state (decoder,
         info.finfo->format, info.width, info.height, self->input_state);
