diff --git a/defaults/defaults b/defaults/defaults
index 6d7184a..a75a6bb 100644
--- a/defaults/defaults
+++ b/defaults/defaults
@@ -45,7 +45,7 @@ shadow_delta_y=-3
 shadow_opacity=50
 show_app_icon=false
 show_dock_shadow=true
-show_frame_shadow=true
+show_frame_shadow=false
 show_popup_shadow=false
 snap_resist=false
 snap_to_border=true
@@ -65,7 +65,7 @@ titleless_maximize=false
 toggle_workspaces=false
 unredirect_overlays=true
 urgent_blink=false
-use_compositing=true
+use_compositing=false
 workspace_count=4
 wrap_cycle=true
 wrap_layout=true
