Index: git/units/systemd-udevd.service.in
===================================================================
--- git.orig/units/systemd-udevd.service.in
+++ git/units/systemd-udevd.service.in
@@ -24,7 +24,7 @@ ExecStart=/lib/systemd/systemd-udevd
 KillMode=mixed
 WatchdogSec=3min
 TasksMax=infinity
-MountFlags=slave
+MountFlags=shared
 MemoryDenyWriteExecute=yes
 RestrictRealtime=yes
 RestrictAddressFamilies=AF_UNIX AF_NETLINK AF_INET AF_INET6
