Index: git/units/emergency.service.in
===================================================================
--- git.orig/units/emergency.service.in
+++ git/units/emergency.service.in
@@ -14,8 +14,8 @@ Conflicts=rescue.service
 Before=shutdown.target
 
 [Service]
-Environment=HOME=@roothomedir@
-WorkingDirectory=-@roothomedir@
+Environment=HOME=/root
+WorkingDirectory=/root
 ExecStart=-@rootlibexecdir@/systemd-sulogin-shell emergency
 Type=idle
 StandardInput=tty-force
Index: git/units/rescue.service.in
===================================================================
--- git.orig/units/rescue.service.in
+++ git/units/rescue.service.in
@@ -14,8 +14,8 @@ After=sysinit.target plymouth-start.serv
 Before=shutdown.target
 
 [Service]
-Environment=HOME=@roothomedir@
-WorkingDirectory=-@roothomedir@
+Environment=HOME=/root
+WorkingDirectory=/root
 ExecStart=-@rootlibexecdir@/systemd-sulogin-shell rescue
 Type=idle
 StandardInput=tty-force

