Index: ./tmpfiles.d/etc.conf.m4
===================================================================
--- ./tmpfiles.d/etc.conf.m4
+++ ./tmpfiles.d/etc.conf.m4
@@ -9,7 +9,7 @@

 L /etc/os-release - - - - ../usr/lib/os-release
 L /etc/localtime - - - - ../usr/share/zoneinfo/UTC
-L+ /etc/mtab - - - - ../proc/self/mounts
+L /etc/mtab - - - - ../proc/self/mounts
 m4_ifdef(`HAVE_SMACK_RUN_LABEL',
 t /etc/mtab - - - - security.SMACK64=_
 )m4_dnl
