From b764b61d09c6980b316e3e408d9c0d6cda0f6518 Mon Sep 17 00:00:00 2001
From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Tue, 10 Jul 2012 15:29:40 +0200
Subject: [PATCH 12/40] v4l2: add support for mpeg4, H.263 and H.264

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2object.c | 38 +++++++++++++++++++++++++++++++++++++-
 1 file changed, 37 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 3ee60ac..da2e38f 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -895,6 +895,15 @@ static const GstV4L2FormatDesc gst_v4l2_formats[] = {
 #ifdef V4L2_PIX_FMT_YVYU
   {V4L2_PIX_FMT_YVYU, TRUE},
 #endif
+#ifdef V4L2_PIX_FMT_MPEG4
+  {V4L2_PIX_FMT_MPEG4, TRUE},
+#endif
+#ifdef V4L2_PIX_FMT_H263
+  {V4L2_PIX_FMT_H263, TRUE},
+#endif
+#ifdef V4L2_PIX_FMT_H264
+  {V4L2_PIX_FMT_H264, TRUE},
+#endif
 };
 
 #define GST_V4L2_FORMAT_COUNT (G_N_ELEMENTS (gst_v4l2_formats))
@@ -1329,6 +1338,21 @@ gst_v4l2_object_v4l2fourcc_to_structure (guint32 fourcc)
       structure = gst_structure_new_empty ("video/x-pwc2");
       break;
 #endif
+#ifdef V4L2_PIX_FMT_MPEG4
+    case V4L2_PIX_FMT_MPEG4:
+      structure = gst_structure_new_empty ("video/mpeg");
+      break;
+#endif
+#ifdef V4L2_PIX_FMT_H263
+    case V4L2_PIX_FMT_H263:
+      structure = gst_structure_new_empty ("video/x-h263");
+      break;
+#endif
+#ifdef V4L2_PIX_FMT_H264
+    case V4L2_PIX_FMT_H264:
+      structure = gst_structure_new_empty ("video/x-h264");
+      break;
+#endif
     default:
       GST_DEBUG ("Unknown fourcc 0x%08x %" GST_FOURCC_FORMAT,
           fourcc, GST_FOURCC_ARGS (fourcc));
@@ -1486,8 +1510,20 @@ gst_v4l2_object_get_caps_info (GstV4l2Object * v4l2object, GstCaps * caps,
 #ifdef V4L2_PIX_FMT_PWC2
     } else if (g_str_equal (mimetype, "video/x-pwc2")) {
       fourcc = V4L2_PIX_FMT_PWC2;
-    }
 #endif
+#ifdef V4L2_PIX_FMT_MPEG4
+    } else if (g_str_equal (mimetype, "video/mpeg")) {
+      fourcc = V4L2_PIX_FMT_MPEG4;
+#endif
+#ifdef V4L2_PIX_FMT_H263
+    } else if (g_str_equal (mimetype, "video/x-h263")) {
+      fourcc = V4L2_PIX_FMT_H263;
+#endif
+#ifdef V4L2_PIX_FMT_H264
+    } else if (g_str_equal (mimetype, "video/x-h264")) {
+      fourcc = V4L2_PIX_FMT_H264;
+#endif
+    }
 
     if (dimensions) {
       const gchar *interlace_mode;
-- 
1.8.3

