From b233c16dc6713f35804b72b6cc716ef1c820a3c4 Mon Sep 17 00:00:00 2001
From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 3 May 2012 16:18:41 +0200
Subject: [PATCH 13/40] v4l2src: set min buffers to 3 if nothing is defined

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2src.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2src.c b/sys/v4l2/gstv4l2src.c
index 00538a7..149d78b 100644
--- a/sys/v4l2/gstv4l2src.c
+++ b/sys/v4l2/gstv4l2src.c
@@ -528,7 +528,7 @@ gst_v4l2src_decide_allocation (GstBaseSrc * bsrc, GstQuery * query)
      * buffer extra to capture while the other two buffers are downstream */
     min += 1;
   } else {
-    min = 2;
+    min = 3;
   }
 
   /* select a pool */
-- 
1.8.3

