From 267a3b17cedf1c62e2569b0031f8c47314a0a836 Mon Sep 17 00:00:00 2001
From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 19 Jul 2012 13:43:31 +0200
Subject: [PATCH 14/40] v4l2src: set max buffers for the pool

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2src.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sys/v4l2/gstv4l2src.c b/sys/v4l2/gstv4l2src.c
index 149d78b..d53b629 100644
--- a/sys/v4l2/gstv4l2src.c
+++ b/sys/v4l2/gstv4l2src.c
@@ -527,8 +527,11 @@ gst_v4l2src_decide_allocation (GstBaseSrc * bsrc, GstQuery * query)
     /* if there is a min-buffers suggestion, use it. We add 1 because we need 1
      * buffer extra to capture while the other two buffers are downstream */
     min += 1;
+    if (!max)
+      max = min;
   } else {
     min = 3;
+    max = 6;
   }
 
   /* select a pool */
-- 
1.8.3

