From c8e20ad4220312fbee5941d91e8b3b2eff618560 Mon Sep 17 00:00:00 2001
From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Wed, 22 Aug 2012 20:30:09 +0200
Subject: [PATCH 34/40] wait for mmaped buffers, when needed

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2bufferpool.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2bufferpool.c b/sys/v4l2/gstv4l2bufferpool.c
index 9893bfa..cf08bc6 100644
--- a/sys/v4l2/gstv4l2bufferpool.c
+++ b/sys/v4l2/gstv4l2bufferpool.c
@@ -426,6 +426,17 @@ gst_v4l2_buffer_pool_start (GstBufferPool * bpool)
       break;
   }
 
+  if (pool->max_buffers != max_buffers) {
+    GstStructure *config;
+    const GstCaps *caps;
+    guint size, min, max;
+
+    config = gst_buffer_pool_get_config (bpool);
+    gst_buffer_pool_config_get_params (config, &caps, &size, &min, &max);
+    gst_buffer_pool_config_set_params (config, caps, size, min, max_buffers);
+    gst_buffer_pool_set_config (bpool, config);
+  }
+
   pool->obj = obj;
   pool->max_buffers = max_buffers;
   pool->buffers = g_new0 (GstBuffer *, max_buffers);
@@ -1148,7 +1159,7 @@ gst_v4l2_buffer_pool_process (GstV4l2BufferPool * pool, GstBuffer * buf)
             if (!start_streaming (pool))
               goto start_failed;
 
-          if (pool->num_queued == pool->num_allocated) {
+          if (pool->num_queued > pool->obj->min_queued) {
             /* all buffers are queued, try to dequeue one and release it back
              * into the pool so that _acquire can get to it again. */
             ret = gst_v4l2_buffer_pool_dqbuf (pool, &to_queue);
-- 
1.8.3

