From 5c616142d1b9d701207640b9473a3df4fe030b99 Mon Sep 17 00:00:00 2001
From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Fri, 26 Oct 2012 14:47:09 +0200
Subject: [PATCH 35/40] v4l2filter: try to keep size when fixating

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 sys/v4l2/gstv4l2filter.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/sys/v4l2/gstv4l2filter.c b/sys/v4l2/gstv4l2filter.c
index c8f5ffe..d43b987 100644
--- a/sys/v4l2/gstv4l2filter.c
+++ b/sys/v4l2/gstv4l2filter.c
@@ -402,8 +402,10 @@ gst_v4l2filter_src_update_caps (GstV4l2Filter * v4l2filter, GstCaps * sinkcaps)
   if (!gst_caps_is_fixed (peercaps)) {
     const GValue *v;
     GstStructure *s1, *s2;
+    GstCaps *caps = gst_caps_copy_nth (peercaps, 0);
 
-    peercaps = gst_caps_truncate (peercaps);
+    gst_caps_unref (peercaps);
+    peercaps = caps;
 
     s1 = gst_caps_get_structure (sinkcaps, 0);
     s2 = gst_caps_get_structure (peercaps, 0);
@@ -414,6 +416,18 @@ gst_v4l2filter_src_update_caps (GstV4l2Filter * v4l2filter, GstCaps * sinkcaps)
       gst_structure_get_fraction (s1, "framerate", &n, &d);
       gst_structure_fixate_field_nearest_fraction (s2, "framerate", n, d);
     }
+    v = gst_structure_get_value (s2, "width");
+    if (v && !gst_value_is_fixed (v)) {
+      gint w;
+      gst_structure_get_int (s1, "width", &w);
+      gst_structure_fixate_field_nearest_int (s2, "width", w);
+    }
+    v = gst_structure_get_value (s2, "height");
+    if (v && !gst_value_is_fixed (v)) {
+      gint w;
+      gst_structure_get_int (s1, "height", &w);
+      gst_structure_fixate_field_nearest_int (s2, "height", w);
+    }
 
     peercaps = gst_caps_fixate (peercaps);
   }
-- 
1.8.3

