Index: git/drivers/char/fsl_otp.h
===================================================================
--- git.orig/drivers/char/fsl_otp.h
+++ git/drivers/char/fsl_otp.h
@@ -28,7 +28,6 @@ static int otp_wait_busy(u32 flags);
 
  /* IMX6 below ============================= */
 
-#include "../../../linux/arch/arm/mach-imx/hardware.h"
 #include <linux/ioport.h>
 #include <linux/platform_device.h>
 #include <linux/regulator/consumer.h>
