diff --git a/src/core/config/embedded_linux.pri b/src/core/config/embedded_linux.pri
index cc8c40f..7ea6d34 100644
--- a/src/core/config/embedded_linux.pri
+++ b/src/core/config/embedded_linux.pri
@@ -50,5 +50,5 @@ GYP_CONFIG += \
     v8_use_snapshot=false \
     want_separate_host_toolset=1 \
 
-contains(QT_CONFIG, system-jpeg): GYP_CONFIG += use_system_libjpeg=1
+contains(QT_CONFIG, system-jpeg): GYP_CONFIG += use_libjpeg-turbo=1
 !contains(QT_CONFIG, pulseaudio): GYP_CONFIG += use_pulseaudio=0
