Index: git/configure
===================================================================
--- git.orig/configure
+++ git/configure
@@ -3207,11 +3207,14 @@ if [ "$XPLATFORM_ANDROID" = "yes" ]; the
         DeviceVar set DEFAULT_ANDROID_NDK_TOOLCHAIN_VERSION "$CFG_DEFAULT_ANDROID_NDK_TOOLCHAIN_VERSION"
 fi
 
+echo "The platform is set to $PLATFORM"
 if [ -d "$PLATFORM" ]; then
+  echo "The platform folder exists"
   QMAKESPEC="$PLATFORM"
 else
   QMAKESPEC="$relpath/mkspecs/${PLATFORM}"
 fi
+echo "The QMAKESPEC is set to $QMAKESPEC"
 if [ -d "$XPLATFORM" ]; then
   XQMAKESPEC="$XPLATFORM"
 else
