@@ -0,0 +1,13 @@
Index: git/src/corelib/io/qurl.h
===================================================================
--- git.orig/src/corelib/io/qurl.h
+++ git/src/corelib/io/qurl.h
@@ -128,7 +128,7 @@ public:
 
     // encoding / toString values
     enum UrlFormattingOption {
-        None = 0x0,
+        Nothing = 0x0,
         RemoveScheme = 0x1,
         RemovePassword = 0x2,
         RemoveUserInfo = RemovePassword | 0x4,

