From 0b9feae76cdc50b5c37a338808cffc3c723177f8 Mon Sep 17 00:00:00 2001
From: Reyhaneh Yazdani <reyhane.y84@gmail.com>
Date: Wed, 7 Dec 2016 11:30:36 +0100
Subject: [PATCH] drm:i915: no DDI and no cherryview condition for DDI display

Signed-off-by: Reyhaneh Yazdani <reyhane.y84@gmail.com>
---
 drivers/gpu/drm/i915/intel_bios.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_bios.c b/drivers/gpu/drm/i915/intel_bios.c
index cf2560708e03..71e2c10f3ee6 100644
--- a/drivers/gpu/drm/i915/intel_bios.c
+++ b/drivers/gpu/drm/i915/intel_bios.c
@@ -1212,7 +1212,7 @@ static void parse_ddi_ports(struct drm_i915_private *dev_priv,
 {
 	enum port port;
 
-	if (!HAS_DDI(dev_priv))
+	if (!HAS_DDI(dev_priv) && !IS_CHERRYVIEW(dev_priv))
 		return;
 
 	if (!dev_priv->vbt.child_dev_num)
-- 
2.12.0

