From a2eec6f84a52523721a22e113669bd0aae918f9a Mon Sep 17 00:00:00 2001
From: Reyhaneh Yazdani <reyhane.y84@gmail.com>
Date: Fri, 18 Nov 2016 13:14:08 +0100
Subject: [PATCH] i915: intel graphic driver mapped to Braswell DDC

Signed-off-by: Reyhaneh Yazdani <reyhane.y84@gmail.com>
---
 drivers/gpu/drm/i915/intel_hdmi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index e6c035b0fc1c..b08cdad25e59 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -2049,7 +2049,8 @@ void intel_hdmi_init_connector(struct intel_digital_port *intel_dig_port,
 		if (IS_BROXTON(dev_priv))
 			intel_hdmi->ddc_bus = GMBUS_PIN_2_BXT;
 		else
-			intel_hdmi->ddc_bus = GMBUS_PIN_DPC;
+			intel_hdmi->ddc_bus = GMBUS_PIN_DPB;
+//			intel_hdmi->ddc_bus = GMBUS_PIN_DPC;
 		intel_encoder->hpd_pin = HPD_PORT_C;
 		break;
 	case PORT_D:
-- 
2.11.0.rc2

